instrumentation/aws-sdk/aws-sdk-1.11/testing/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AbstractS3ClientTest.java

```diff
@@ -0,0 +1,200 @@
+                new ClientConfiguration()
+                    .withRequestTimeout(50 /* ms */)
+                    .withRetryPolicy(
+                        PredefinedRetryPolicies.getDefaultRetryPolicyWithCustomMaxRetries(1)))
+            .withEndpointConfiguration(
+                new AwsClientBuilder.EndpointConfiguration(
+                    server.httpUri().toString(), "us-east-1"))
+            .build();
+
+    Throwable caught = catchThrowable(() -> client.getObject("someBucket", "someKey"));
+    assertThat(caught).isInstanceOf(AmazonClientException.class);
+    assertThat(Span.current().getSpanContext().isValid()).isFalse();
+
+    testing()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> {
+                      span.hasName("S3.GetObject")
+                          .hasKind(CLIENT)
+                          .hasStatus(StatusData.error())
+                          .hasNoParent()
+                          .hasAttributesSatisfyingExactly(
+                              equalTo(URL_FULL, server.httpUri().toString()),
+                              equalTo(HTTP_REQUEST_METHOD, "GET"),
+                              equalTo(SERVER_PORT, server.httpPort()),
+                              equalTo(SERVER_ADDRESS, "127.0.0.1"),
+                              equalTo(RPC_SYSTEM, "aws-api"),
+                              equalTo(RPC_SERVICE, "Amazon S3"),
+                              equalTo(RPC_METHOD, "GetObject"),
+                              equalTo(stringKey("aws.endpoint"), server.httpUri().toString()),
+                              equalTo(stringKey("aws.agent"), "java-aws-sdk"),
+                              equalTo(stringKey("aws.bucket.name"), "someBucket"),
+                              satisfies(
+                                  ERROR_TYPE,
+                                  val ->
+                                      val.satisfiesAnyOf(
+                                          v ->
+                                              assertThat(v)
+                                                  .isEqualTo(SdkClientException.class.getName()),
+                                          v ->
+                                              assertThat(v)
+                                                  .isEqualTo(
+                                                      AmazonClientException.class.getName()))));
+
+                      try {
+                        span.hasException(
+                            new AmazonClientException("Unable to execute HTTP request"));
+                      } catch (AssertionError e) {
+                        span.hasException(
+                            new SdkClientException(
+                                "Unable to execute HTTP request: Request did not complete before the request timeout configuration."));
+                      }
+                    }));
```

Comment:
Do you know when it gets which exception? If the exception depends on whether it is regular or latest dep test then we could simplify this and the `ERROR_TYPE` assertion.

Reply:
seems like its always SdkClientException, so I was able to simplify. great catch!